analyses/transcriptomic-dimension-reduction/05-seq-center-mitochondrial-genes.Rmd

```diff
@@ -0,0 +1,225 @@
+        legend.direction = "vertical",
+        legend.text = element_text(size = rel(0.5)),
+        legend.title = element_text(size = rel(0.5)),
+        legend.key.size = unit(0.25, "cm")
+  )) 
+
+# figure settings
+knitr::opts_chunk$set(fig.width = 8)
+```
+
+
+Define directories and file names: 
+```{r}
+# Directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+tumor_dir <- file.path(root_dir, "analyses", "tumor-purity-exploration")
+umap_dir <- file.path(
+  root_dir,
+  "analyses",
+  "transcriptomic-dimension-reduction", 
+  "results"
+)
+palette_dir <- file.path(root_dir, "figures", "palettes")
+
+# UMAP files with different samples:
+rsem_umap <- file.path(umap_dir, "rsem_stranded_log_umap_scores_aligned.tsv")
+rsem_no_mito_umap <- file.path(umap_dir, "rsem_stranded_no-mito_log_umap_scores_aligned.tsv")
+
+# palette mapping file
+pal_file <- file.path(palette_dir,  "broad_histology_cancer_group_palette.tsv")
+
+# FPKM
+expression_file <- file.path(data_dir, "pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds")
+
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+```
+
+```{r}
+# Read in palette data
+palette_mapping_df <- readr::read_tsv(pal_file) %>%
+  dplyr::select(broad_histology, broad_histology_display, broad_histology_hex, 
+                cancer_group, cancer_group_display, cancer_group_hex) 
+
+# Read in and prep UMAP data
+# helper function
+readr_prep_umap <- function(filename, 
+                            pal_df = palette_mapping_df) {
+  readr::read_tsv(filename) %>% 
+    dplyr::rename(UMAP1 = X1, UMAP2 = X2) %>%
+    dplyr::inner_join(pal_df) %>%
+    dplyr::mutate(seq_center = forcats::fct_relevel(seq_center, 
+                                                    "BGI", "BGI@CHOP Genome Center", "NantOmics"))
+}
+
+umap_all <- readr_prep_umap(rsem_umap)
+umap_no_mito <- readr_prep_umap(rsem_no_mito_umap) 
+
+# Read in expression and convert to data frame
+expression_df <- readr::read_rds(expression_file) %>%
+  tibble::as_tibble(rownames = "gene_symbol")
+
+# Read in metadata
+metadata_df <- readr::read_tsv(metadata_file)
+```
+
+
+## Explore mitochondrial genes
+
+For mitochondrial genes, do we see expression patterns for BGI@CHOP samples consistent with those posted in the issue linked above?
+
+```{r}
+# First, which diagnoses are sequenced at BGI@CHOP?
+relevant_groups <- metadata_df %>%
+  dplyr::filter(RNA_library == "stranded", 
+                stringr::str_starts(seq_center, "BGI")) %>%
+  tidyr::drop_na(broad_histology) %>%
+  dplyr::pull(broad_histology) %>%
+  unique()
+
+# Data frame of log2(FPKM) of mitochondrial genes for relevant diagnoses
+mito_seq_center_df <- expression_df %>%
+  dplyr::filter(stringr::str_starts(gene_symbol, "MT-")) %>%
+  tidyr::gather("Kids_First_Biospecimen_ID", 
+                "fpkm", 
+                tidyselect::starts_with("BS_")) %>%
+  dplyr::mutate(log2_fpkm = log2(fpkm+1)) %>%
```

Comment:
Just want to note my minor discomfort with `+1` as a zero-correction for FPKM, but it does not really matter in this case.

Reply:
For better or worse, this is all up in OpenPBTA :/